jwst/tweakreg/tweakreg_catalog.py

```diff
@@ -73,10 +78,17 @@ def make_tweakreg_catalog(model, kernel_fwhm, snr_threshold, sharplo=0.2,
                       dqflags.pixel['DO_NOT_USE']) &
                      model.dq).astype(bool)
 
-    bkg = JWSTBackground(model.data, box_size=bkg_boxsize,
-                         coverage_mask=coverage_mask)
+    columns = ['id', 'xcentroid', 'ycentroid', 'flux']
+    try:
+        bkg = JWSTBackground(model.data, box_size=bkg_boxsize,
+                             coverage_mask=coverage_mask)
+        threshold_img = bkg.background + (snr_threshold * bkg.background_rms)
+    except ValueError as e:
+        log.warning(f"Error determining sky background: {e.args[0]}")
+        # return an empty catalog
+        catalog = Table(names=columns, dtype=(int, float, float, float))
+        return catalog
```

Comment:
```suggestion
    try:
        bkg = JWSTBackground(model.data, box_size=bkg_boxsize,
                             coverage_mask=coverage_mask)
    except ValueError as e:
        log.warning(f"Error determining sky background: {e.args[0]}")
        # return an empty catalog
        catalog = Table(names=columns, dtype=(int, float, float, float))
        return catalog
        
    threshold_img = bkg.background + (snr_threshold * bkg.background_rms)
```

Reply:
That approach of only including line 83 (bkg = JWSTBackground()) in the `try` block doesn't work, because that line only creates an instance of the "bkg" object. It's line 91 (threshold_img = bkg.background ....) that actually throws the error and hence it needs to be in the `try` block.  Trust me, I tried it that way first and it still crashed.

Reply:
Ah, OK.

Reply:
Yes, the background is not actually calculated until the `bkg.background` (lazy)property is requested.